File: frontend/src/types/productCategory.ts
export interface ProductCategory {
  id: number
  code: string
  name: string
  description: string | null
  isActive: boolean
  createdAtUtc: string
  updatedAtUtc: string | null
}

export interface CreateProductCategoryRequest {
  name: string
  description: string | null
}

export interface UpdateProductCategoryRequest {
  name: string
  description: string | null
  isActive: boolean
}

export type ProductCategoryFormValues = UpdateProductCategoryRequest

File: backend/Project1.Api/Entities/ProductCategory.cs
namespace Project1.Api.Entities;

public sealed class ProductCategory
{
    public int Id { get; set; }

    public string Code { get; set; } = string.Empty;

    public string Name { get; set; } = string.Empty;

    public string? Description { get; set; }

    public bool IsActive { get; set; } = true;

    public DateTimeOffset CreatedAtUtc { get; set; } = DateTimeOffset.UtcNow;

    public DateTimeOffset? UpdatedAtUtc { get; set; }

    public ICollection<Product> Products { get; set; } = [];
}
